metadata-ingestion/tests/integration/sql_server/setup/setup.sql

```diff
@@ -159,4 +159,16 @@ EXEC sp_attach_schedule
 GO
 EXEC dbo.sp_add_jobserver
     @job_name = N'Weekly Demo Data Backup'
-GO
\ No newline at end of file
+GO
+
+DROP DATABASE IF EXISTS [DB_WITH@SPEC_SYMB];
```

Comment:
we can add a comment mentioning that this database is for testing `quote_schemas` and `use_odbc` in `metadata-ingestion/tests/integration/sql_server/source_files/mssql_special_symbols_db.yml`


In the remaining recipes, in some cases we are adding `schema_pattern `and in others `database_pattern` and `schema_pattern`
* can be consistent?
* would `database_pattern.deny: [ 'DB_WITH@SPEC_SYMB' ]` be enough?